src/controllers/user.js

```diff
@@ -74,4 +77,64 @@ const signin = (req, res) => {
   
 }
 
-module.exports = { signup, signin};
\ No newline at end of file
+export const getAllUsers = async (req, res) => {
+
+    const user = await User.findAll();
+    if (user){
+      return res.status(200).json({user});
+    }else{
+      res.status(500).send({message: 'Error'});
+    }
+        
+}
+
+
+export const getUserById = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const user = await User.findByPk(id);
+
+    if (user) {
+      return res.status(200).json({ user });
+    }else{
+      return res.status(404).json({message: 'No User with the specified'});
+    }
+
+  }catch(error){
+    res.status(500).send({ message: 'Error'})
+  }
+ 
+}
+
+export const updateUserById = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const [user] = await User.update(req.body, {where: {id : id }});
+    
+    if(user){
+      return res.status(200).json({ message: 'User updated successfully' });
+    }else{
+      return res.send({ message: `Cannot update User with id=${id}. User not found`});
+    }
+
+  }catch(error){
+    return res.status(500).send({ message:'Error'});
+  }
+}
```

Comment:
You need a kind of validation while updating because I found that two user can have one email in database.. and also make sure a user can update his/her own account except admin who can have that privilege to all users.

Reply:
Added a validation to check if the submitted email equals to an existing email in the db
`export const updateUserById = async (req, res) => {
  
    const users = await User.findAll();
    for(let i=0; i < users.length; i++){
      if (users[i].email === req.body.email){
          return res.status(409).send({
             message: 'User update failed, a user with the specified email exist',
          });
      }
    }
    try {
      const id = req.params.id;
      const [user] = await User.update(req.body, {where: {id : id }});
      
      if(user){
        return res.status(200).json({ message: 'User updated successfully' });
      }else{
        return res.send({ message: `Cannot update User with id=${id}. User not found`});
      }
  
    }catch(error){
      return res.status(500).send({ message:'Error'});
    }

}`